### PR TITLE
refactor: Switch to hooks

### PR DESCRIPTION
**Pull request by rschristian**
## Description

Converts the existing class components into functional components with hooks.

## Reason for Change

Most people in the Preact/React communities these days will be writing functional components with hooks. As such, this template should probably use them. 

If this is a user's first experience with Preact, I'd argue using newer syntax is better. However, maybe one of each should be kept, to show that Preact supports both syntaxes? Not sure myself, depends on what maintainers want. 

### Changelog
* [x]  Convert existing class components (`App` and `Profile`) to use hooks
* [x]  Updated an import (`import style from './style';` -> `import style from './style.css';`) purely for consistency

### CODE REVIEW

File: template/src/components/app.js [hunk 1]
```diff
@@ -1,32 +1,21 @@
...
 
-export default class App extends Component {
-	
-	/** Gets fired when the route changes.
-	 *	@param {Object} event		"change" event from [preact-router](http://git.io/preact-router)
-	 *	@param {string} event.url	The newly routed URL
-	 */
-	handleRoute = e => {
-		this.currentUrl = e.url;
-	};
```
Comment (rschristian) [inline template/src/components/app.js:18]: I'm not entirely sure why this was added to begin with, as it was not being used in any particular way, so it was removed. I can add it back if it does serve some purpose.

File: template/src/routes/profile/index.js [hunk 2]
```diff
@@ -1,47 +1,30 @@
...
-				<p>
-					<button onClick={this.increment}>Click Me</button>
-					{' '}
-					Clicked {count} times.
-				</p>
-			</div>
-		);
-	}
+import { h } from 'preact';
+import {useEffect, useState} from "preact/hooks";
```
Comment (prateekbh) [inline template/src/routes/profile/index.js:2]: We might not want to begin with anything beyond whats in core.

Comment (rschristian) [reply template/src/routes/profile/index.js:2]: Is there any particular reason for this, maybe a long-term maintenance sort of thing? Fair enough of course whatever the reason.

Comment (prateekbh) [reply template/src/routes/profile/index.js:2]: I imagine it was the additional weight that we didn't wanted the base template to have. Hooks is an optional feature which is not free.

I'd also like to revisit this decision with @ForsakenHarmony @reznord nad @marvinhagemeister

Comment (ForsakenHarmony) [reply template/src/routes/profile/index.js:2]: I think hooks is something most people want, so probably not a problem

File: template/src/routes/profile/index.js [hunk 3]
```diff
@@ -1,47 +1,30 @@
...
+import style from './style.css';
+
+// Note: `user` comes from the URL, courtesy of our router
+const Profile = ({ user }) => {
+	const [time, setTime] = useState(Date.now());
+	const [count, setCount] = useState(10);
+
+	useEffect(() => {
+		setInterval(() => setTime(Date.now()), 1000);
```
Comment (ForsakenHarmony) [inline template/src/routes/profile/index.js]: this is missing the cleanup now

```suggestion
		let timer = setInterval(() => setTime(Date.now()), 1000);
		return () => clearInterval(timer);
```

File: template/src/routes/profile/index.js [hunk 4]
```diff
@@ -1,47 +1,30 @@
...
+
+	return (
+		<div class={style.profile}>
+			<h1>Profile: {user}</h1>
+			<p>This is the user profile for a user named { user }.</p>
+
+			<div>Current time: {new Date(time).toLocaleString()}</div>
+
+			<p>
+				<button onClick={() => setCount((count) => ++count)}>Click Me</button>
```
Comment (ForsakenHarmony) [inline template/src/routes/profile/index.js]:
```suggestion
				<button onClick={() => setCount((count) => count + 1)}>Click Me</button>
```

I know it works both ways, but I feel this is the cleaner version

**Review by ForsakenHarmony — APPROVED** [review]

**Comment by ForsakenHarmony** [comment]
🙌

File: template/src/routes/profile/index.js (hunk 3 above)
Comment (rschristian) [reply template/src/routes/profile/index.js]: Whoops, sorry about that.

**Pull request merged**